package/upgrade/upgrade_manifests.sh

```diff
@@ -38,19 +142,44 @@ upgrade_rancher()
     return
   fi
 
+  # Wait for Rancher to settle down before start upgrading, just in case
+  wait_capi_cluster fleet-local local 0
+  pre_generation=$(kubectl get clusters.cluster.x-k8s.io local -n fleet-local -o=jsonpath="{.status.observedGeneration}")
+
+  # XXX Workaround for https://github.com/rancher/rancher/issues/36914
+  # Delete all rancher's clusterrepos so they will be updated by the new version rancher pods
+  # Note: The leader pod would create these cluster repos
+  kubectl delete clusterrepos.catalog.cattle.io rancher-charts
+  kubectl delete clusterrepos.catalog.cattle.io rancher-rke2-charts
+  kubectl delete clusterrepos.catalog.cattle.io rancher-partner-charts
+  kubectl delete settings.management.cattle.io chart-default-branch
+
   REPO_RANCHER_VERSION=$REPO_RANCHER_VERSION yq -e e '.rancherImageTag = strenv(REPO_RANCHER_VERSION)' values.yaml -i
-  ./helm upgrade rancher ./*.tgz --namespace cattle-system -f values.yaml
+  ./helm upgrade rancher ./*.tgz --namespace cattle-system -f values.yaml --wait
 
   # Wait until new version ready
   until [ "$(get_running_rancher_version)" = "$REPO_RANCHER_VERSION" ]
   do
     echo "Wait for Rancher to be upgraded..."
     sleep 5
   done
+
+  echo "Wait for Rancher dependencies helm releases..."
+  wait_helm_release cattle-fleet-system fleet $REPO_FLEET_CHART_VERSION $REPO_FLEET_APP_VERSION deployed
+  wait_helm_release cattle-fleet-system fleet-crd $REPO_FLEET_CRD_CHART_VERSION $REPO_FLEET_CRD_APP_VERSION deployed
+  wait_helm_release cattle-system rancher-webhook $REPO_RANCHER_WEBHOOK_CHART_VERSION $REPO_RANCHER_WEBHOOK_APP_VERSION deployed
```

Comment:
```suggestion
  wait_helm_release cattle-fleet-system fleet fleet-$REPO_FLEET_CHART_VERSION $REPO_FLEET_APP_VERSION deployed
  wait_helm_release cattle-fleet-system fleet-crd fleet-crd-$REPO_FLEET_CRD_CHART_VERSION $REPO_FLEET_CRD_APP_VERSION deployed
  wait_helm_release cattle-system rancher-webhook rancher-webhook-$REPO_RANCHER_WEBHOOK_CHART_VERSION $REPO_RANCHER_WEBHOOK_APP_VERSION deployed
```

https://github.com/harvester/harvester-installer/pull/252 output versions in harvester-release.yaml without chart name, would be nice to prefix the chart name here.